RFCs/0002-add-pdk-config.md

```diff
@@ -109,9 +111,35 @@ Ref: (Verb-Noun Reference)[https://docs.google.com/document/d/1zX0FJBAvAIK3d3L3Q
       - If `<value>` is already present in the value list, issues a notice that value was already on list, outputs the
         current values of `<key>`, and exits with a success status.
 
-    If `<key>` is a user-defined configuration key where we cannot infer whether or not it can have multiple values,
-    assume it is a single-value key unless the user includes the `--add` flag. If the user includes the `--add` flag,
-    treat it as a multi-value key as described above.
+    If the `--force` flag is set then the `<key>` will be literally set to `<value>`.  That is, list modifications like above will not be used.
+    This flag is useful when resetting configuration to a known good state, or when automating the PDK to configure projects on first use.
+
+    If `<key>` is a configuration key where we cannot infer what type of value it should be, the user can include the `--type` (aliased with `--as`) flag
+    to indicate what type of value it should be. The included types are those common to JSON and YAML documents (String, Number, Boolean, Array, Hash and Null)
+
+      - `--type number`. Treats the value as a number. For example `1`, `1.0`, `-1.0`
+
+      - `--type boolean`. Treats the value as a boolean (True/False). For example `true`, `TRUE`, `False`, `yes`, `no`, `0` (False), `-1` (True)
+
+      - `--type array`. Treats the value as a array element.  For example `value` would become an array with a single string element of value, that is `["value"]`
+
+      - `--type array`. If given no value, treats the value as an empty array, that is `[]`
+
+      - `--type empty|nil|null|nul`. Must be given no value and treats the value as empty (or `nil` in Ruby parlance)
```

Comment:
Is this just another way to unset something? Or is there another use case?

Reply:
The use case here is that JSON specifies a `nul` case.

``` json
{ something: nul }
```

which is different than it doesn't exist.

But I think I know what you're getting at.  Perhaps, for now, I add a comment that it's a future usecase?

I _think_ (can't be sure off the top of my head) that the PDK::Config does not differentiate between `nil|nul|empty|undefined` and `missing|not-set`.  But I can see a time in the future when that may be needed.

Reply:
OK, yeah I would say maybe for now we don't need to implement but can in the future if we find a use case for it.

Reply:
Done.  I'll mark it as Possible Future Usecase!

Reply:
Moved to Future Considerations section of the document